### PR TITLE
save selected code, subject in App.tsx to enhence content component

### LINKED ISSUE #51: Use the subject and code to make requests to /subject and /seats
Currently the API calls to `/sections` (to get sections) and `/seats` etc. have to wait for the call to `getCourseDetails` to finish in order to make requests. This is because the course details use `pid` and the sections/seats use `subject`, `code` to make their requests however we only keep track of the `pid` and `term`. 

We can make the app faster by storing the `subject` and `code` just like the `pid`. This information is currently available since the sidebar needs it to display information.

### PR DESCRIPTION
**Pull request by chsunwoo1002**
## Overview

Closes #51 Use the subject and code to make requests to /subject and /seats 

1. Create hooks for currently selected code, subject by using state
2. pass the function to sidebar.tsx
3. get selected code and subject values in sidebar component
4. use these values to the content component to rendering /section and /seat faster.

### CODE REVIEW

**Comment by aomi** [comment]
@chsunwoo1002 I don't think we want to reset the selected course when the user navigates back to the subjects list.

**Review by aomi — COMMENTED** [review]
Minor things and 1 point to just discuss. We might want to combine some states together into a object though (but I think that can be another PR).

File: src/app/content/Content.tsx [hunk 1]
```diff
@@ -19,29 +19,37 @@ export interface ContentProps {
...
               code={data.code}
               title={data.title}
               description={data.description || ''}
               credits={data.credits.value}
             />
-            <SectionsContainer term={term} subject={data.subject} code={data.code} />
-          </>
-        )}
+          )}
+          <SectionsContainer term={term} subject={subject} code={code} />
```
[the remark this answers is not in the document]
Comment (keithradford) [reply src/app/content/Content.tsx:51]: Yeah, I agree. It's small but just makes sense.

[the remark this answers is not in the document]
Comment (chsunwoo1002) [reply src/app/content/Content.tsx:51]: so, move SectionsContainer component into the bracket? and using <> to make it one element?

[the remark this answers is not in the document]
Comment (aomi) [reply src/app/content/Content.tsx:51]: Leave it as is and just remove the <>

[the remark this answers is not in the document]
Comment (chsunwoo1002) [reply src/app/content/Content.tsx]: @aomi Can u do reviewing? Please let me know if I need to change anything.

**Review by aomi — APPROVED** [review]

**Pull request merged**